std/string.d

```diff
@@ -2920,14 +2923,68 @@ if (isConvertibleToString!Range)
     assert(testAliasedString!stripLeft("  hello"));
 }
 
+/// Ditto
+auto stripLeft(Range, Char)(Range input, const(Char)[] chars)
+if (((isForwardRange!Range && isSomeChar!(ElementEncodingType!Range)) ||
+     isConvertibleToString!Range) && isSomeChar!Char)
+{
+    static if (isConvertibleToString!Range)
+        return stripLeft!(StringTypeOf!Range)(input, chars);
+    else
+    {
+        for (; !input.empty; input.popFront)
```

Comment:
Why don't you use `foreach` here?
In this case there's _any_ performance penalty of this abstraction.
https://github.com/dlang/phobos/pull/6092#issuecomment-361397749

Reply:
I think the catch here is that we need to iterate *from the back*, not from the front. So we'll either have to use `foreach_reverse` (*ick*), or `std.range.retro`, which then would require passing the range in by reference, since we need to guarantee that the range is actually modified in order to return it to the user. So it'll end up being something ugly like this:
````
import std.range : retro;
foreach (ch; (&input).retro) { ... }
````
which is equally *ick*.

Reply:
Oh wait, what I said only applies to the `stripRight` variant.  Here we're moving forwards, so there's no `foreach_reverse` or `retro` needed.

I'd say the `for` loop here is justified (using foreach would require passing the range by ref, just in case, so it still entails the ugly `&range` syntax).  Perhaps a more self-documenting way to write this would be:
````
while (!input.empty && chars.indexOf(input.front) != -1)
    input.popFront;
````
which, at least to me, reads more intuitively as "while the range is not empty and the front element is a character to be stripped, pop the front element".

Reply:
Well, it boils down to whether we want to do auto-decoding or not ;-)